runtime-modules/content/src/lib.rs

```diff
@@ -2781,6 +2790,87 @@ impl<T: Trait> Module<T> {
             .cloned()
             .collect::<BTreeSet<_>>()
     }
+
+    fn ensure_sufficient_balance_for_channel_transfer(
+        owner: &ChannelOwner<T::MemberId, T::CuratorGroupId>,
+        transfer_cost: BalanceOf<T>,
+    ) -> DispatchResult {
+        let balance = match owner {
+            ChannelOwner::Member(member_id) => {
+                let controller_account_id =
+                    T::MemberAuthenticator::controller_account_id(*member_id)?;
+
+                Balances::<T>::usable_balance(&controller_account_id)
+            }
+            ChannelOwner::CuratorGroup(_) => T::ContentWorkingGroup::get_budget(),
+        };
+
+        ensure!(
+            balance >= transfer_cost,
+            Error::<T>::InsufficientBalanceForTransfer
+        );
+        Ok(())
+    }
+
+    // Validates channel transfer acceptance parameters: commitment params, new owner balance.
+    fn validate_channel_transfer_acceptance(
+        actor: &ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+        commitment_params: &TransferParameters<T::MemberId, BalanceOf<T>>,
+        params: &PendingTransfer<T::MemberId, T::CuratorGroupId, BalanceOf<T>>,
+    ) -> DispatchResult {
+        ensure!(
+            params.transfer_params == *commitment_params,
+            Error::<T>::InvalidChannelTransferCommitmentParams
+        );
+
+        // Check for new owner balance only if the transfer is not free.
+        if !params.transfer_params.is_free_of_charge() {
+            let owner = actor_to_channel_owner::<T>(actor)?;
+
+            Self::ensure_sufficient_balance_for_channel_transfer(
+                &owner,
+                params.transfer_params.price,
+            )?;
+        }
+
+        Ok(())
+    }
+    // Transfers balance from the new channel owner to the old channel owner.
+    fn pay_for_channel_swap(
+        old_owner: &ChannelOwner<T::MemberId, T::CuratorGroupId>,
+        new_owner: &ChannelOwner<T::MemberId, T::CuratorGroupId>,
+        price: BalanceOf<T>,
+    ) -> DispatchResult {
+        // Decrease (or slash) balance for the new owner
+        match new_owner {
+            ChannelOwner::Member(member_id) => {
+                let controller_account_id =
+                    T::MemberAuthenticator::controller_account_id(*member_id)?;
+
+                let _ = Balances::<T>::slash(&controller_account_id, price);
+            }
+            ChannelOwner::CuratorGroup(_) => {
+                let new_budget = T::ContentWorkingGroup::get_budget().saturating_sub(price);
```

Comment:
Are we sure this budget is sufficient, what if it is empty or too low?

Reply:
Ah seeing now that you check this prior, can you add comments here which describe that sufficiency checks have been done prior, and also add `debug_asserts`? I know we only use debug asserts for invariant that are across extrinsic calls, but just feels so important to make sure these two parts of the codepath do not eventually drift apart in some weird scenario.

Reply:
I added a comment. However, `debug_assert` burdens the code when it's ready and invariants are checked in tests.

Reply:
I think the utility of this is beyond the verification, its for clarity for reader, to highlight less visible invariants, but I rest my case ;)